### PR TITLE
CSHARP-4985: Verify that operands to numeric operators in LINQ expressions are represented as numbers on the server.

### PR DESCRIPTION
**Pull request by rstam**
Let's not try and push this to master until after I get back. Feel free to take a look and comment. Mainly I wanted to get this work backed up.

I probably need a few more tests (though there are some tests already).

### CODE REVIEW

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
A variety of LINQ3 tests are timing out with these changes. Please investigate the root cause of the timeouts before we review the code closely. Thanks.

**Comment by rstam** [comment]
> A variety of LINQ3 tests are timing out with these changes. Please investigate the root cause of the timeouts before we review the code closely.

Done. Ready for review.

**Comment by rstam** [comment]
There were a bunch of failed variants with cryptic log files that made no sense to me. I'm hoping they were transient Evergreen failures so I've rebase on master and force pushed the branch to trigger a new patch build.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Misc/SerializationHelper.cs [hunk 1]
```diff
@@ -106,11 +113,52 @@ public static BsonType GetRepresentation(IBsonSerializer serializer)
...
             return SerializationHelper.GetRepresentation(serializer) == BsonType.Document;
         }
 
+        public static bool IsRepresentedAsInteger(IBsonSerializer serializer)
+        {
+            var representation = GetRepresentation(serializer);
+            return IsIntegerRepresentation(representation);
+        }
+
+        public static bool IsRepresentedAsIntegerOrNullableInteger(AggregationExpression translation)
```
Comment (JamesKovacs) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Misc/SerializationHelper.cs:145]: All other `IsRepresentedAsXyz` methods take an `IBsonSerializer` as the only argument. Consider the same here.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Minor improvement suggested below.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Misc/SerializationHelper.cs (hunk 1 above)
Comment (rstam) [reply src/MongoDB.Driver/Linq/Linq3Implementation/Misc/SerializationHelper.cs:145]: There already is an overload of `IRepresentedAsIntegerOrNullableInteger` that takes an `IBsonSerializer`.

This is a convenience method when checking that a translation is represented as an integer or nullable integer.

It is analogous to another similar convenience method like `EnsureRepresentationIsNumeric`.

Let me know if this changes your request.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**